lxd/device/cdi/spec.go

```diff
@@ -62,15 +63,26 @@ func generateNvidiaSpec(cdiID ID, inst instance.Instance) (*specs.Spec, error) {
 	}
 
 	rootPath := ""
-	if shared.InSnap() {
+	devRootPath := ""
+	if s.OS.InUbuntuCore() {
+		//
+		// This magic "gpu-2404-2" name comes from:
+		// https://github.com/canonical/mesa-2404/blob/0e48b4d1b8e5cb4d3098d64417025824decd9846/scripts/bin/gpu-2404-provider-wrapper.in#L5
+		//
+		// Also, we can't use `/snap/lxd/current/gpu-2404-2` as a rootPath because of a bug in nvcdi package.
```

Comment:
is there a link to the bug in question you could add here?

Reply:
> is there a link to the bug in question you could add here?

there is no buglink. That's what I found during my investigation of nvcdi internals.

Reply:
OK thanks, would it be worth opening an issue on their Github page?

Reply:
> OK thanks, would it be worth opening an issue on their Github page?

yeah, definitely, but I decided to go with a quick and simple workaround from LXD side at first.

I'll fill an issue or even put up a PR for nvidia thing a bit later.